.github/workflows/nightly.yml

```diff
@@ -0,0 +1,229 @@
+      run: |
+       tox -e flake8
+
+###############################################
+################## CPU-BUILD ##################
+###############################################
+  build-cpu:
+    runs-on: ${{ matrix.os }}
+    needs: static-analysis
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python: [3.6]
+        # different kind of tests are located in tests/<unit|integration|smoke> folders
+        test-kind: ['unit', 'smoke']
+        # pytest markers configured in tox.ini. See https://docs.pytest.org/en/6.2.x/example/markers.html
+        test-marker: ['not spark and not gpu']
+        include:
+          # Integration tests needs a more powermachine with more memory. GitHub-hosted agents only have 7GB memory.
+          - os: self-hosted
+            python: 3.6
+            test-kind: 'integration'
+            test-marker: 'not spark and not gpu'
+
+    steps:
+    - uses: actions/checkout@v2
+    ################# Run Python tests #################
+    - name: Use Python ${{ matrix.python }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python }}
+    # There are little documentation on how to call **local** actions
+    # https://docs.github.com/en/actions/creating-actions/about-actions#choosing-a-location-for-your-action
+    # but there is some working insights from this discussion:
+    # https://github.community/t/path-to-action-in-the-same-repository-as-workflow/16952/2
+    - name: Run ${{ matrix.test-kind }} tests ('${{ matrix.test-marker }}')
+      uses: ./.github/workflows/actions/run-tests
+      with:
+        test-kind: ${{ matrix.test-kind }}
+        test-marker: ${{ matrix.test-marker }}
+    # Currently GitHub workflow cannot call an action from another action
+    # https://github.community/t/call-an-action-from-another-action/17524
+    # Else this shared step can also be move to the local action: .github/workflows/actions/run-tests
+    - name: Upload Code Coverage
+      uses: actions/upload-artifact@v2
+      with:
+        name: code-cov
+        path: .coverage*
+
+###############################################
+################# SPARK-BUILD #################
+###############################################
+  build-spark:
+    runs-on: ${{ matrix.os }}
+    needs: static-analysis
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        java: [8]
+        python: [3.6]
+        # different kind of tests are located in tests/<unit|integration|smoke> folders
+        test-kind: ['unit', 'smoke'] 
+        # pytest markers configured in tox.ini. See https://docs.pytest.org/en/6.2.x/example/markers.html
+        test-marker: ['spark and not gpu']
+        include:
+          # Integration tests needs a more powermachine with more memory. GitHub-hosted agents only have 7GB memory.
+          - os: self-hosted
+            python: 3.6
+            test-kind: 'integration'
+            test-marker: 'not spark and not gpu'
```

Comment:
Not sure if this should be `spark and not gpu`? This is the spark build

Reply:
good catch :D